<launch>
    
    <node  name="msf_position_pose_sensor1" pkg="msf_updates" type="position_pose_sensor" clear_params="true" output="screen">			
        <remap from="msf_core/imu_state_input" to="/fcu/imu" />
        <rosparam file="$(find msf_updates)/params/demo1_VH/msf_sensor_fix.yaml"/>
        <remap from="msf_updates/position_input" to="/noise_drift_handler1/position_output" />
        <remap from="msf_updates/pose_with_covariance_input" to="noise_drift_handler1/pose_output" />
        <param name="capability_group" value="MSF" />
    </node>    

    <node name="rovio1" pkg="rovio" type="rovio_node" output="screen">
        <param name="filter_config" value="$(find rovio)/cfg/rovio.info"/>
        <param name="camera0_config" value="$(find rovio)/cfg/euroc_cam0_equidist.yaml"/>
        <param name="camera1_config" value="$(find rovio)/cfg/euroc_cam1_equidist.yaml"/>
    	<remap from="cam0/image_raw" to="cam0/image_raw"/>
		<remap from="cam1/image_raw" to="cam1/image_raw"/>
    	<remap from="imu0" to="rovio_divergence_handler1/imu0_output"/>
    </node>
    
    <node name="pos_to_error1" pkg="msf_updates" type="pos_to_error.py" output="screen">
        <remap from="pos_to_error1/datainput" to="/msf_position_pose_sensor1/msf_core/state_out"/>
        <remap from="pos_to_error1/truthinput" to="/vicon/firefly_sbx/firefly_sbx"/>
        <rosparam file="$(find msf_updates)/params/demo1_VH/pos_to_error_fix.yaml"/>
    </node>

    <node name="noise_drift_handler1" pkg="msf_updates" type="noise_drift_handler.py" output="screen">
        <remap from="noise_drift_handler1/pose_input" to="/rovio/pose_with_covariance_stamped"/>
        <remap from="noise_drift_handler1/transform_input" to="/vicon/firefly_sbx/firefly_sbx"/>
        <rosparam file="$(find msf_updates)/params/demo1_VH/noise_drift_handler_fix.yaml"/>
    </node>
    
    <node name="rovio_divergence_handler1" pkg="msf_updates" type="rovio_divergence_handler.py" output="screen">
        <remap from="rovio_divergence_handler1/cam0_input" to="/cam0/image_raw"/>
        <remap from="rovio_divergence_handler1/cam1_input" to="/cam1/image_raw"/>
        <remap from="rovio_divergence_handler1/imu0_input" to="/imu0"/>
        <rosparam file="$(find msf_updates)/params/demo1_VH/rovio_divergence_handler_fix.yaml"/>
    </node>
</launch>
